### ISSUE TITLE
Extension modules compiled with MinGW64 on windows are exporting symbols

### ISSUE THREAD

**Issue by Keithcat1** [Nuitka/Nuitka#925]
I thought this was fixed in a previous version of Nuitka. This only seems to be happening with modules compiled with GCC, not modules compiled by MSVC.

**Comment by kayhayen** [comment]
Can you provide more details please?

**Comment by Keithcat1** [comment]
When I say exporting symbols, I mean that the extension module DLL is exporting a bunch of what seems to be Nuitka internal symbols.
If you run dumpbin/exports module.pyd, it will show a lot of symbols besides PyInit

**Comment by kayhayen** [comment]
I can confirm this to be true. I was using gcc 10 and winlibs, my initial suspect was this being about old gcc, but clearly not the case.

**Comment by kayhayen** [comment]
I just pushed a fix for this to factory. I don't think this was ever any better with MinGW64. The new `clang.exe` support should inherit the fix as well.

Also while being there, I patched Scons to allow the implib to be inside the build folder rather than near the file to be created, avoiding the littering that has annoyed me in the past.

This is likely to make it into 0.6.11 still, which is in final touches.
